EncountifyAPI/Controllers/VisitedLocationsController.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using EncountifyAPI.Data;
+using EncountifyAPI.Models;
+using System.Data.SqlClient;
+using Microsoft.Extensions.Configuration;
+using System.Text;
+using System.Data;
+using System.Diagnostics;
+
+namespace EncountifyAPI.Controllers
+{
+    [Route("API/[controller]")]
+    [ApiController]
+    public class VisitedLocationsController : ControllerBase
+    {
+        private readonly string ConnectionString;
+
+        public VisitedLocationsController(IConfiguration configuration)
+        {
+            ConnectionString = configuration.GetConnectionString("EncountifyAPIContext");
+        }
+
+        /// <summary>
+        /// Get all visited locations
+        /// </summary>
+        [HttpGet]
+        public IEnumerable<VisitedLocation> GetAllVisitedLocations()
+        {
+            return ExecuteVisitedLocationReader("SELECT * FROM VisitedLocations");
+        }
+
+        /// <summary>
+        /// Get visited location by Id
+        /// </summary>
+        [HttpGet("{id}")]
+        public IEnumerable<VisitedLocation> GetVisitedLocation(int id)
+        {
+            return ExecuteVisitedLocationReader("SELECT * FROM VisitedLocations WHERE Id = @id", id: id);
+        }
+
+        /// <summary>
+        /// Get user's visited locations
```

Comment:
It seems that comments in line 48 and 57 are mixed up

Reply:
Oh, I actually forgot to change the routes :D